src/Setup/UpdateVersions.purs

```diff
@@ -34,141 +34,171 @@ import Node.FS.Sync (writeTextFile)
-                        , repo.name
+      Right [] -> pure Nothing
+      Right objects ->
+        Just
+          <$> Array.catMaybes
+          <$> for objects \obj ->
+            case obj .: "tag_name" of
+              Left e ->
+                throwError $ error $ fold
+                  [ "Failed to get tag from GitHub response: "
+                  , printJsonDecodeError e
+                  ]
+              Right tagName ->
+                case tagStrToVersion tagName of
+                  Left _ -> do
+                    liftEffect $ warning $ fold
+                      [ "Got invalid version"
+                      , tagName
+                      , " from "
+                      , repo.name
+                      ]
+                    pure Nothing
+                  Right version -> case obj .: "draft" of
+                    Left e ->
+                      throwError $ error $ fold
+                        [ "Failed to get draft from GitHub response: "
+                        , printJsonDecodeError e
                         ]
-                      pure Nothing
-                    Right version -> case obj .: "draft" of
-                      Left e ->
-                        throwError $ error $ fold
-                          [ "Failed to get draft from GitHub response: "
-                          , printJsonDecodeError e
-                          ]
-                      Right isDraft ->
-                        pure
-                          if isDraft then Nothing
-                          else Just version
-
-  tagStrToVersion tagStr =
-    tagStr
-      # String.stripPrefix (String.Pattern "v")
-      # fromMaybe tagStr
-      # Version.parseVersion
-
-  -- If a tool doesn't use GitHub releases and instead only tags versions, then
-  -- we have to fetch the tags, parse them as appropriate versions, and then sort
-  -- them according to their semantic version to get the latest one.
-  fetchFromGitHubTags repo = recover do
-    let url = "https://api.github.com/repos/" <> repo.owner <> "/" <> repo.name <> "/tags"
-
-    AX.get RF.json url >>= case _ of
-      Left err -> do
-        throwError (error $ AX.printError err)
-
-      Right { body } -> case traverse (_ .: "name") =<< decodeJson body of
-        Left e -> do
-          throwError $ error $ fold
-            [ "Failed to decode GitHub response. This is most likely due to a timeout.\n\n"
-            , printJsonDecodeError e
-            , stringifyWithIndent 2 body
-            ]
-
-        Right arr -> do
-          let
-            tags = Array.mapMaybe (tagStrToVersion >>> hush) arr
-
-          case maximum tags of
-            Nothing ->
-              throwError $ error "Could not download latest release version."
-
-            Just v ->
-              pure v
+                    Right isDraft ->
+                      pure
+                        if isDraft then Nothing
+                        else Just version
```

Comment:
This is a little clunky (I know it's not your code) -- mind updating it to use `guard`?

Reply:
I tried, but either implementation of `guard` doesn't work. 
```
src/Setup/UpdateVersions.purs:165:23

  165                        guard isDraft
                             ^^^^^^^^^^^^^
  
  No type class instance was found for
  
    Control.Alternative.Alternative Aff
  
  while applying a function guard
    of type Alternative t2 => Boolean -> t2 Unit
    to argument isDraft
  while checking that expression guard isDraft
    has type t0 t1
  in value declaration toolVersions
```
and 
```
src/Setup/UpdateVersions.purs:165:23

  165                        guard isDraft version
                             ^^^^^^^^^^^^^^^^^^^^^
  
  Could not match type
  
    Version
  
  with type
  
    t0 (Maybe t3)
```
or
```
src/Setup/UpdateVersions.purs:165:23

  165                        guard isDraft $ Just version
                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  
  Could not match type
  
    Version
  
  with type
  
    Maybe t3
```

Reply:
Ah, the ol' `guardA`. No worries, we can leave it as-is.